superset-frontend/src/dashboard/components/nativeFilters/ChartCustomization/ChartCustomizationForm.tsx

```diff
@@ -0,0 +1,1147 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { FC, useEffect, useMemo, useState, useRef, useCallback } from 'react';
+import { t, styled, css, SLOW_DEBOUNCE, useTheme } from '@superset-ui/core';
+import { debounce } from 'lodash';
+import { Form, FormItem } from 'src/components/Form';
+import { Input, TextArea } from 'src/components/Input';
+import { Radio } from 'src/components/Radio';
+import Select from 'src/components/Select/Select';
+import Collapse from 'src/components/Collapse';
+import { InfoTooltipWithTrigger } from '@superset-ui/chart-controls';
+import Loading from 'src/components/Loading';
+import { getChartDataRequest } from 'src/components/Chart/chartAction';
+import { CollapsibleControl } from '../FiltersConfigModal/FiltersConfigForm/CollapsibleControl';
+import { ColumnSelect } from '../FiltersConfigModal/FiltersConfigForm/ColumnSelect';
+import DatasetSelect from '../FiltersConfigModal/FiltersConfigForm/DatasetSelect';
+import DefaultValue from '../FiltersConfigModal/FiltersConfigForm/DefaultValue';
+import { ChartCustomizationItem } from './types';
+import { getFormData } from '../utils';
+
+const StyledForm = styled(Form)`
+  .ant-form-item {
+    margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
+  }
+`;
+
+const StyledContainer = styled.div`
+  ${({ theme }) => `
+   display: flex;
+   flex-direction: row;
+   gap: ${theme.gridUnit * 4}px;
+   padding: ${theme.gridUnit * 2}px;
+ `}
+`;
+
+const FORM_ITEM_WIDTH = 300;
+
+const StyledFormItem = styled(FormItem)`
+  width: ${FORM_ITEM_WIDTH}px;
+  margin-bottom: ${({ theme }) => theme.gridUnit * 4}px;
+
+  .ant-form-item-label > label {
+    font-size: ${({ theme }) => theme.typography.sizes.m}px;
+    font-weight: ${({ theme }) => theme.typography.weights.normal};
+    color: ${({ theme }) => theme.colors.grayscale.dark1};
+  }
```

Comment:
Not going to comment on these next but the same logic applies everywhere. Can we remove as many style customizations as possible and go vanilla?